### PR TITLE
Adding UI tests for the  Basic Menu settings. 🎲 !!!!

### PR DESCRIPTION
**Pull request by KMaragh**
---

### Pull Request checklist

- [X ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by AaronMT** [comment]
Did you want to re-base and try again? I think there was something wrong with the devices at the time of execution.

**Comment by KMaragh** [comment]
Will make an update to this today :). I will rebase and try again

File: app/src/androidTest/java/org/mozilla/fenix/helpers/matchers/BitmapDrawableMatcher.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (currentDrawable is StateListDrawable) {
+            currentDrawable = currentDrawable.current
+        }
+        if (currentDrawable is BitmapDrawable) {
+            val bitmap = currentDrawable.bitmap
+            return bitmap.sameAs(otherBitmap)
+        }
+        return false
+    }
+}
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/helpers/matchers/BitmapDrawableMatcher.kt]: What exactly does this helper do?

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 2]
```diff
@@ -86,45 +92,63 @@ class SettingsBasicsTest {
...
-        // Open 3dot (main) menu
-        // Select settings
-        // Select "Search engine"
-        // Toggle 'Show search suggestions' to 'off'
-        // Back arrow twice to home screen
-        // Enter: "mozilla" in navigation bar
-        // Verify no suggestions provided
+        homeScreen {
+        }.openNavigationToolbar {
+            verifySuggestionsAreMoreThan(1, "mozilla")
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: I think we should include "search" in the function name, `verifySearchSuggestions...`

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 3]
```diff
@@ -86,45 +92,63 @@ class SettingsBasicsTest {
...
+        }.goBack {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            disableShowSearchSuggestions()
+        }.goBack {
+        }.goBack {
+        }.openNavigationToolbar {
+            verifySuggestionsAreEqualTo(0, "mozilla")
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: "Search" again

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 4]
```diff
@@ -86,45 +92,63 @@ class SettingsBasicsTest {
...
-        // Open 3dot (main) menu
-        // Enter navigation bar and verify visited sites appear
-        // Verify bookmarks exist
-        // Open 3dot (main) menu
-        // Select settings
-        // Select "Search engine"
-        // Toggle off "Show visited sites and bookmarks"
-        // Back arrow twice to home screen
-        // Verify history and bookmarks are gone
+        val webPage = getGenericAsset(mockWebServer, 1)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: Let's use `page1`, `page2`, `page3`

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 5]
```diff
@@ -86,45 +92,63 @@ class SettingsBasicsTest {
...
+        val secondWebPage = getGenericAsset(mockWebServer, 2)
+        val thirdWebPage = getGenericAsset(mockWebServer, 3)
+
+        homeScreen {
+        }.openNavigationToolbar {
+        }.enterURLAndEnterToBrowser(webPage.url) {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: let's remember to include verification after each visit 
`verifyPageContent(page1.content)`

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 6]
```diff
@@ -141,40 +165,45 @@ class SettingsBasicsTest {
...
-        // Open static website in navigation bar
-        // Verify that text is now at 180%
-        // Select settings
-        // Select Accessibility
-        // Toggle font size back to 'off'
-        // Verify that "Font Size" header, description, slider bar and sample text all disappear
+        val fenixApp = activityIntentTestRule.activity.applicationContext as FenixApplication
+        val webpage = getLoremIpsumAsset(mockWebServer).url
+
+        val textSizePercentage = 180
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: Let's include a comment about this value

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 7]
```diff
@@ -48,6 +53,7 @@ class HomeScreenRobot {
     fun verifyHomeWordmark() = assertHomeWordmark()
     fun verifyHomeToolbar() = assertHomeToolbar()
     fun verifyHomeComponent() = assertHomeComponent()
+    fun verifyDefaultSearchEngineIsDDG() = verifyDDGSearchEngineIcon()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: I think we should pass in a search engine to verify a default. The default could change in product and we would have to rewrite this function.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 8]
```diff
@@ -202,95 +221,108 @@ private fun assertHomeToolbar() = onView(ViewMatchers.withResourceName("toolbar"
...
+
+private fun verifySearchEngineIcon(searchEngineIcon: Bitmap, searchEngineName: String) {
+    onView(withId(R.id.search_engine_icon))
+        .check(matches(withBitmapDrawable(searchEngineIcon, searchEngineName)))
+}
+
+private fun getSearchEngine(searchEngineName: String) =
+    Search(appContext).searchEngineManager.getDefaultSearchEngine(appContext, searchEngineName)
+
+private fun verifyDDGSearchEngineIcon() {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: Yeah again not the biggest fan of these super specific per search engine checks. Can they be re-written to be more generic incase the engines change in product?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/NavigationToolbarRobot.kt [hunk 9]
```diff
@@ -10,26 +10,45 @@ import android.net.Uri
...
+import org.mozilla.fenix.helpers.assertions.AwesomeBarAssertion.Companion.suggestionsAreGreaterThan
 
 /**
  * Implementation of Robot Pattern for the URL toolbar.
  */
 class NavigationToolbarRobot {
+
+    fun verifySuggestionsAreMoreThan(suggestionSize: Int, searchTerm: String) = assertSuggestionsAreMoreThan(suggestionSize, searchTerm)
+    fun verifySuggestionsAreEqualTo(suggestionSize: Int, searchTerm: String) = assertSuggestionsAreEqualTo(suggestionSize, searchTerm)
+    fun verifyHistoryAndBookmarksAreGone() = assertHistoryAndBookmarksAreGone()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/NavigationToolbarRobot.kt]: `verifyNoHistoryBookmarks`

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 10]
```diff
@@ -66,7 +66,12 @@ class SearchRobot {
 
     fun clickDuckDuckGoResult() {
         mDevice.waitNotNull(Until.findObjects(By.text("DuckDuckGo")), TestAssetHelper.waitingTime)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt]: Similar to above, is it possible to make this one more generic?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 11]
```diff
@@ -116,20 +122,27 @@ private fun allowPermissionButton(): UiObject {
...
+        TestAssetHelper.waitingTime
+    )
     return onView(allOf(withId(R.id.searchScanButton)))
 }
 
 private fun clearButton() = onView(withId(R.id.mozac_browser_toolbar_clear_view))
 
 private fun assertDuckDuckGoURL() {
-    mDevice.waitNotNull(Until.findObject(By.textContains("https://duckduckgo.com/?q=mozilla")), TestAssetHelper.waitingTime)
+    mDevice.waitNotNull(
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt]: Same as above

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 12]
```diff
@@ -116,20 +122,27 @@ private fun allowPermissionButton(): UiObject {
...
+        Until.findObject(By.textContains("https://duckduckgo.com/?q=mozilla")),
+        TestAssetHelper.waitingTime
+    )
     onView(allOf(withText(startsWith("https://duckduckgo.com"))))
         .check(matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
 }
 
 private fun assertDuckDuckGoResults() {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt]: same as above

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuDefaultBrowserRobot.kt [hunk 13]
```diff
@@ -8,18 +8,37 @@ package org.mozilla.fenix.ui.robots
...
  * Implementation of Robot Pattern for the settings DefaultBrowser sub menu.
  */
 class SettingsSubMenuDefaultBrowserRobot {
 
+    companion object {
+        const val DEFAULT_APPS_SETTINGS_ACTION = "android.settings.MANAGE_DEFAULT_APPS_SETTINGS"
+    }
+
     fun verifyOpenLinksInPrivateTab() = assertOpenLinksInPrivateTab()
+    fun verifyDefaultBrowserIsOff() = assertDefaultBrowserIsOff()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuDefaultBrowserRobot.kt]: "Disabled"

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 14]
```diff
@@ -29,7 +33,7 @@ class SettingsBasicsTest {
     private lateinit var mockWebServer: MockWebServer
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: Lets include comments for what the test does. See some of the other tests.

File: app/src/androidTest/java/org/mozilla/fenix/helpers/matchers/BitmapDrawableMatcher.kt (hunk 1 above)
Comment (KMaragh) [reply app/src/androidTest/java/org/mozilla/fenix/helpers/matchers/BitmapDrawableMatcher.kt]: It is used to get the underlying bitmap from an Imageview. There are certain instances where the drawable in an imageview is of type StateListDrawable. This type of object can have multiple drawables assigned to it, and this method is used to get the current drawables bitmap.

**Comment by KMaragh** [comment]
Updated some text in the accessibility settings.

**Pull request merged**